packages/flutter/lib/src/cupertino/search_field.dart

```diff
@@ -0,0 +1,403 @@
+      _createLocalController();
+    }
+  }
+
+  @override
+  void didUpdateWidget(CupertinoSearchTextField oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.controller == null && oldWidget.controller != null) {
+      _createLocalController(oldWidget.controller!.value);
+    } else if (widget.controller != null && oldWidget.controller == null) {
+      unregisterFromRestoration(_controller!);
+      _controller!.dispose();
+      _controller = null;
+    }
+  }
+
+  @override
+  void restoreState(RestorationBucket? oldBucket, bool initialRestore) {
+    if (_controller != null) {
+      _registerController();
+    }
+  }
+
+  void _registerController() {
+    assert(_controller != null);
+    registerForRestoration(_controller!, 'controller');
+  }
+
+  void _createLocalController([TextEditingValue? value]) {
+    assert(_controller == null);
+    _controller = value == null
+        ? RestorableTextEditingController()
+        : RestorableTextEditingController.fromValue(value);
+    if (!restorePending) {
+      _registerController();
+    }
+  }
+
+  @override
+  String? get restorationId => widget.restorationId;
+
+  VoidCallback _getDefaultOnSuffixTap() {
+    return () {
+      final bool textChanged = _effectiveController.text.isNotEmpty;
+      _effectiveController.clear();
+      if (widget.onChanged != null && textChanged)
+        widget.onChanged!(_effectiveController.text);
+    };
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final String placeholder = widget.placeholder ??
+        CupertinoLocalizations.of(context)?.searchTextFieldPlaceholerLabel ??
+        'Search';
+
+    final TextStyle placeholderStyle = widget.placeholderStyle ??
+        const TextStyle(color: CupertinoColors.secondaryLabel);
+
+    // The icon size will be scaled by a factor of the accessibility text scale,
+    // to follow the behavior of `UISearchTextField`.
+    final double scaledIconSize =
+        MediaQuery.textScaleFactorOf(context) * widget.itemSize;
+
+    // If decoration was not provided, create a decoration with the provided
+    // background color and border radius.
+    final BoxDecoration decoration = widget.decoration ??
+        BoxDecoration(
+          color: widget.backgroundColor ?? CupertinoColors.tertiarySystemFill,
+          borderRadius: widget.borderRadius ?? _kDefaultBorderRadius,
+        );
+
+    final IconThemeData iconThemeData = IconThemeData(
+        color: CupertinoDynamicColor.resolve(widget.itemColor, context),
+        size: scaledIconSize);
+
+    final Widget prefix = Padding(
+      child: IconTheme(
+          child: const Icon(CupertinoIcons.search), data: iconThemeData),
+      padding: widget.prefixInsets,
+    );
+
+    final Widget suffix = Padding(
+      child: CupertinoButton(
+        child: IconTheme(child: widget.suffixIcon, data: iconThemeData),
+        onPressed: widget.onSuffixTap ?? _getDefaultOnSuffixTap(),
+        minSize: 0.0,
```

Comment:
Why does the `minSize` have to set to 0? The button should always be larger than 44 px so it's easier to tap right?

Reply:
Per our conversation on discord, we will leave this issue for a future PR.